### PR TITLE
fix: sql optimizations to speed up various tx queries

### PR DESCRIPTION
**Pull request by zone117x**
* Add sql index to speed up tx list queries.
* Remove redundant sql queries for already found tx db data.
* Make the max pg connection pool limit configurable via env var.

### CODE REVIEW

File: src/api/routes/tx.ts [hunk 1]
```diff
@@ -75,7 +75,7 @@ export function createTxRouter(db: DataStore): RouterWithAsync {
 
     // TODO: use getBlockWithMetadata or similar to avoid transaction integrity issues from lazy resolving block tx data (primarily the contract-call ABI data)
     const results = await Bluebird.mapSeries(txResults, async tx => {
-      const txQuery = await getTxFromDataStore(db, { txId: tx.tx_id, includeUnanchored });
+      const txQuery = await getTxFromDataStore(db, { txId: tx.tx_id, dbTx: tx, includeUnanchored });
```
Comment (rafaelcr) [inline src/api/routes/tx.ts:78]: Shouldn't this be `includeUnanchored: includeUnanchored` ?

Comment (zone117x) [reply src/api/routes/tx.ts:78]: As opposed to the shorthand of `includeUnanchored`? It's only a stylistic difference. I don't feel strongly either way

Comment (rafaelcr) [reply src/api/routes/tx.ts:78]: Oh, I think I just hadn't seen that shorthand before 👍 all good

**Review by rafaelcr — APPROVED** [review]
Looks great

**Review by wileyj — APPROVED** [review]
shipit

**Pull request merged**